atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/WriteInfoPartitioner.java

```diff
@@ -58,22 +58,21 @@ public WriteInfoPartitioner(KeyValueService kvs, Supplier<Integer> numShards) {
      * Filters out all writes made into tables with SweepStrategy NOTHING, then partitions the writes according to
      * shard, strategy, and start timestamp of the transaction that performed the write.
      */
-    public Map<PartitionInfo, List<WriteInfo>> filterAndPartition(List<WriteInfo> writes) {
-        return partitionWritesByShardStrategyTimestamp(filterOutUnsweepableTables(writes));
-    }
-
-    @VisibleForTesting
-    List<WriteInfo> filterOutUnsweepableTables(List<WriteInfo> writes) {
-        return writes.stream()
-                .filter(writeInfo -> getStrategy(writeInfo).isPresent())
-                .collect(Collectors.toList());
+    public PartitionedWriteInfo filterAndPartition(List<WriteInfo> writes) {
+        Preconditions.checkArgument(!writes.isEmpty(), "There must be at least one write.");
```

Comment:
I found it confusing to follow that the partitioner does the filtering as well. Can we move this logic out and also rename this class to `SweepableWriteInfoPartitioner..` or something of that sort